bulk_site_creator/views.py

```diff
@@ -240,13 +274,35 @@ def create_bulk_job(request: HttpRequest) -> Optional[JsonResponse]:
             workflow_state="pending",
         )
 
+        log_extra = {
+            'sis_account_id': sis_account_id,
+            'user_id': user_id,
+            'user_full_name': user_full_name,
+            'user_email': user_email,
+            'school': school_id,
+            'term_id': term_id,
+            'term_name': term_name,
+            'department_id': department_id,
+            'department_name': department_name,
+            'course_group_id': course_group_id,
+            'course_group_name': course_group_name,
+            'template_id': template_id
+        }
+        # Sanitized input for log statements.
+        term_id = str(term_id)
+        sis_account_id = str(sis_account_id)
+        logger.debug(f'Generating task objects for term ID {term_id} (term name {term_name}) '
+                     f'and custom Canvas account sis ID {sis_account_id}.', extra={'log_extra': log_extra})
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It looks like you are trying to use `QuerySet.extra` for constructing raw SQL queries using external inputs. This could make your code vulnerable to SQL injection attacks, which can lead to loss of critical data. We recommend you to use `params` or `select_params` for passing parameters to the query string.

[Learn more](https://cwe.mitre.org/data/definitions/89.html)

Similar issue at line numbers 305, 318, and 321.